src/services/webclient.ts

```diff
@@ -2008,12 +2008,39 @@ export class WebClientService {
             .then((titlePrefix) =>  {
                 const title = `${titlePrefix} ${senderName}`;
                 let body = '';
-                switch (message.type) {
+                let messageType = message.type;
+                let caption = message.caption;
+                let captionString = '';
+                if (typeof caption !== 'undefined') {
+                    captionString = captionString + ': ' + caption;
+                }
+                let messageTypeString = this.$translate.instant('messageTypes.' + messageType);
+                switch (messageType) {
                     case 'text':
                         body = message.body;
                         break;
+                    case 'location':
+                        body = messageTypeString + ': ' + message.location.poi;
+                        break;
+                    case 'file':
+                        if (message.file.type === 'image/gif') {
+                            body = this.$translate.instant('messageTypes.' + 'gif') + captionString;
+                            break;
+                        }
+                        // Display caption, if available otherwise use filename
+                        if (captionString.length > 0) {
+                            body = messageTypeString + captionString;
+                            break;
+                        }
+                        body = messageTypeString + ': ' + message.file.name;
```

Comment:
Since this is an if-else situation, it would probably be clearer if you put this part of the logic into an `else` block. Then you can also remove the `break` online 233.